### PR TITLE
chore: Replace eventBus with mitt.js [CW-3275]

### PR DESCRIPTION
**Pull request by fayazara**
# Replace the deprecated `eventBus` with mitt.js

## Description

Since eventBus and it's respective methods are deprecated and removed from all future releases of vue, this was blocking us from migrating. This PR replaces eventBus with [mitt](https://github.com/developit/mitt). I have created a wrapper mitt.js to simulate the same old event names so it's backwards compatible, without making a lot of changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

1. Made sure all the places we're listening to bus events are working as expected.
2. Respective specsf or the events from mitt.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by fayazara** [comment]
Note for self : Get more info on -
`app/javascript/dashboard/components/ui/DatePicker/DatePicker.vue` - Vue 3 script syntax

`app/javascript/v3/components/SnackBar/Container.vue`

File: app/javascript/dashboard/components/ui/DatePicker/DatePicker.vue
Comment (scmmishra) [inline app/javascript/dashboard/components/ui/DatePicker/DatePicker.vue:1]: Let's replace these also with `showAlert`

**Review by scmmishra — CHANGES_REQUESTED** [review]
One tiny change, the rest is good to go.

**Review by scmmishra — APPROVED** [review]

**Review by scmmishra — CHANGES_REQUESTED** [review]
@fayazara there are instances where we have potential infintie loops, where `showAlert` is already a function defined in the component, and then the mixin is added which calls `showAlert` again. PTAL

CleanShot 2024-05-30 at 10 26 50@2x

**Comment by fayazara** [comment]
@scmmishra this was a really good catch. Some of these methods were doing more than just showing alerts, hence renamed them to showAlertMessage.

CleanShot 2024-05-31 at 00 32 36@2x

**Review by scmmishra — APPROVED** [review]

**Review by muhsin-k — APPROVED** [review]

**Pull request merged**